src/renderer/atlas/AtlasEngine.cpp

```diff
@@ -251,6 +251,16 @@ try
 {
     _flushBufferLine();
 
+    // PaintCursor() is only called when the cursor is visible, but we need to invalidate the cursor area
+    // even if it isn't. Otherwise a transition from a visible to an invisible cursor wouldn't be rendered.

```

Comment:
The confusing part here (at least for me) and what led to this bug, is that there's a distinction in our code between a visible cursor and a cursor that's "on". `!IsCursorVisible()` forces the cursor to not be drawn, but it can still be on/off when it blinks while being invisible.
...and while it blinks it still gets invalidated (why tho?).

BTW there's also `IsConversionArea()` which is supposed to be an invisibility flag, but it got broken at some point in conhost's rewrite to C++ so now it _is_ visible (whoops) but unlike `!IsCursorVisible()` it doesn't invalidate the area being drawn, so it just happens to work (unless you invalidate the cursor).
As mentioned yesterday, the cursor code is not in a great shape right now. 🥲